src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyTool.java

```diff
@@ -690,16 +704,57 @@ public void overrideLicenses(LicenseMap licenseMap, SortedMap<String, MavenProje
                 // empty license means not fill, skip it
                 continue;
             }
-            getLogger().info("overriding for " + project + ", " + Arrays.toString(licenses));
-            licenseMap.removeProject( project );
 
-            // add license in map
-            addLicense( licenseMap, project, licenses );
+            projects.forEach(project -> {
+                getLogger().info("overriding for " + project + ", " + Arrays.toString(licenses));
+                licenseMap.removeProject( project );
+
+                // add license in map
+                addLicense( licenseMap, project, licenses );
+            });
 
         }
         licenseMap.removeEmptyLicenses();
     }
 
+    public List<MavenProject> getProjectFromCustomOverrideFile(String id, SortedMap<String, MavenProject> artifactCache, VersionRangeFactory versionRangeFactory, NumericVersionFactory numericVersionFactory) {
+        String[] overrideProjectGAV = id.split("--");
+
+        if (overrideProjectGAV.length != 3) {
+            return Collections.emptyList();
+        }
+
+        String overrideGroupId = overrideProjectGAV[0];
+        String overrideArtifactId = overrideProjectGAV[1];
+        String overrideVersionId = overrideProjectGAV[2];
+
+        // Ensure versionIdRanges is in the correct format ( must start and end with [ or ( and ) or ] )
+        String versionIdRanges = overrideVersionId.matches("^[\\[(].*[\\])]$") ? overrideVersionId : "[" + overrideVersionId + "]";
+
+        // Compile the project name pattern (group and artifact ID)
+        String projectNamePatternString = String.format("^%s--%s--.*$", overrideGroupId, overrideArtifactId);
+        Pattern projectNamePattern = Pattern.compile(projectNamePatternString);
+
+        return artifactCache.entrySet().stream()
+                .filter(entry -> projectNamePattern.matcher(entry.getKey()).matches())
```

Comment:
What do we need pattern here? not just equals?

Reply:
The entry key format in artifactCache always includes scalar versions, whereas the override project format can include version ranges. The pattern is used to ensure that the group and artifact IDs match, regardless of the version declaration format used.

Reply:
I suggest to split the key, it will work faster